src/mod/features/ev_iv_ui.cpp

```diff
@@ -0,0 +1,297 @@
+        customColorFields.r = 0.323143f;
+        customColorFields.g = 0.023153f;
+        customColorFields.b = 0.01096f;
+    }
+    else if (power <= 19 && power >= 10) {
+        //99251AFF
+        customColorFields.r = 0.318547f;
+        customColorFields.g = 0.0185f;
+        customColorFields.b = 0.01033f;
+    }
+    else
+    {
+        //99241AFF
+        customColorFields.r = 0.318547f;
+        customColorFields.g = 0.017642f;
+        customColorFields.b = 0.01033f;
+    }
+}
+
+HOOK_DEFINE_REPLACE(displayPower1) {
+    static void Callback(Dpr::UI::BoxStatusPanelUI::Object *boxStatusPanel) {
+        int32_t power = boxStatusPanel->fields.pokemonParam->GetPower(POWER_ID_ATK);
+        Dpr::Message::MessageWordSetHelper::SetDigitWord(0, power);
+    }
+};
+
+void displayPower2(Dpr::UI::BoxStatusPanelUI::Object *boxStatusPanel)
+{
+    int32_t power = boxStatusPanel->fields.pokemonParam->GetPower(POWER_ID_DEF);
+    Dpr::Message::MessageWordSetHelper::SetDigitWord(0, power);
+}
+
+HOOK_DEFINE_REPLACE(displaySummaryPower) {
+    static void Callback(Dpr::UI::StatusPanelUI::Object *statusPanel) {
+        Dpr::UI::StatusPanelLocals::Object *locals = statusPanel->fields.locals;
+        Pml::PokePara::CoreParam *pokemonParam = locals->fields.pokemonParam;
+        System::Array<int32_t> *_powerIdMap = locals->fields.__4__this->fields._powerIdMap;
+        int32_t i = statusPanel->fields.i;
+        int32_t power = pokemonParam->GetPower(_powerIdMap->m_Items[i]);
+
+        for (int j = 0; j < statusPanel->fields.locals->fields.__4__this->fields._colorUpDowns->max_length; j++)
+        {
+            mapColorToPower(power);
+
+            statusPanel->fields.locals->fields.__4__this->fields._colorUpDowns->m_Items[j].fields.r = customColorFields.r;
+            statusPanel->fields.locals->fields.__4__this->fields._colorUpDowns->m_Items[j].fields.g = customColorFields.g;
+            statusPanel->fields.locals->fields.__4__this->fields._colorUpDowns->m_Items[j].fields.b = customColorFields.b;
+            statusPanel->fields.locals->fields.__4__this->fields._colorUpDowns->m_Items[j].fields.a = customColorFields.a;
+        }
+
+        Dpr::Message::MessageWordSetHelper::SetDigitWord(0, power);
+    }
+};
+
+Il2CppObject * thunk_FUN_7100252fd8(void * typeInfo, void * data);
+
+HOOK_DEFINE_REPLACE(Dpr_UI_BoxStatusPanel_GetJudgeTextCode) {
+    static System::String *
+    Callback(Dpr::UI::BoxStatusPanelUI::Object *__this, Pml::PokePara::CoreParam *pokemonParam, int32_t powerId) {
+        
+        bool trainingDone = pokemonParam->IsTrainingDone(powerId);
+        int32_t messageID;
+        if (trainingDone) {
+            messageID = 0x241;
+        } else {
+            messageID = 0x24e + pokemonParam->GetTalentPower(powerId);
+        }
+
+        //system_load_typeinfo(0x2609);
+
+        Il2CppObject *intObject = thunk_FUN_7100252fd8(int_TypeInfo, (void *) &messageID);
+        return System::String::Concat(intObject);
```

Comment:
For this method here, I think it would be a lot cleaner to re-write it. Doing something like calling the MessageManager to get the proper strings ("Hyper Trained!", which is SS_box_577 in ss_box) using GetSimpleMessage at address 710210d000 could work. As for getting the number, instead of using this weird thunk, you can just use c++'s std::to_string on the int and then use that in System::String::Create.